src/nvim/api/buffer.c

```diff
@@ -1161,6 +1255,12 @@ Boolean nvim_buf_set_mark(Buffer buffer, String name, Integer line, Integer col,
     return res;
   }
 
+  // load buffer first if it's not loaded
```

Comment:
I'm unsure if named marks make use of the memline (the part of the buffer which is loaded/unloaded) or not. would need to check the implementation of `set_mark()`